components/add_users_to_team/add_users_to_team.test.jsx

```diff
@@ -28,7 +28,7 @@ describe('components/AddUsersToTeam', () => {
         const getProfilesNotInTeam = jest.fn().mockResolvedValue({data: true});
         const actions = {...baseActions, getProfilesNotInTeam};
         const props = {...baseProps, actions};
-        const wrapper = shallow(
+        const wrapper = shallowWithIntl(
```

Comment:
This isn't something to change in this PR, but I didn't realize how removing usage of  `localizeMessage` would make it so that we have to do this throughout the tests. We might want to think about if there's a less invasive of doing this in the future.

Reply:
Maybe it is stupid, but I have one idea. Let's say we have the following code in the `BackstageHeader` component:
```JSX
<i
    className='fa fa-angle-right'
    title={formatMessage({id: 'generic_icons.breadcrumb', defaultMessage: 'Breadcrumb Icon'})}
/>
```
We need the `intl` object either from injectIntl or from the context in the component to the `formatMessage` method. Cannot we instead create a component that would be used only for rendering icons with translated `title` and `aria-label` attributes? Code of this component might look like this:
```JSX
export default class TranslatedIcon extends React.PureComponent {
    static propTypes = {
        className: PropTypes.string.isRequired,
        messageId: PropTypes.string.isRequired,
        defaultMessage: PropTypes.string.isRequired,
    };

    render() {
        const {className, messageId, defaultMessage} = this.props;
        return (
            <FormattedMessage
                id={messageId}
                defaultMessage={defaultMessage}
            >
                {(title) => (
                    <i
                        className={className}
                        title={title}
                    />
                )}
            </FormattedMessage>
        );
    }
}
```
Usage of the `TranslatedIcon` in the `BackstageHeader` component might look like this:
```JSX
<TranslatedIcon
    className='fa fa-angle-right'
    messageId='generic_icons.breadcrumb'
    defaultMessage='Breadcrumb Icon'
/>
```
By using the above solution we have 2 advantages:
1. The `BackstageHeader` component doesn't need the `intl` object.
2. We don't need to use the `shallowWithIntl` method in `BackstageHeader` component tests. We just can use the `shallow` method and make sure that proper properties were provided to the `TranslatedIcon` component.

@hmhealey what do you think about this?

Reply:
That would work as long as we are only using `shallow` instead of `mount` since I think `mount` would still attempt to render the `TranslatedIcon` which would cause the same error. There is also the option of just waiting until react-intl releases a new version and switches to using the new Context API since I think that will allow us to set a default value without faking it like we do here. Either way, I don't think it's a big deal right now, but it would be a nice improvement to make in the future.

Regardless of that though, I think `TranslatedIcon` could be a nice thing to include anyway. While a lot of these icons are re-used all over the place, there might be some icons that we use only once, so creating a new component would be overkill. It could also be used to make the implementations of the re-used icons like `SearchIcon` even simpler as well.

Reply:
Also, just a thought, but you could make that implementation a bit simpler. Since the `FormattedMessage` only cares about the `messageId` and `defaultMessage` props, you could just pass everything else into the `i` tag like this to make it easier to add other props like event handlers without the `TranslatedIcon` needing to know what those might be.

```jsx
export default class TranslatedIcon extends React.PureComponent {
    static propTypes = {
        messageId: PropTypes.string.isRequired,
        defaultMessage: PropTypes.string.isRequired,
    };

    render() {
        const {messageId, defaultMessage, ...props} = this.props;
        return (
            <FormattedMessage
                id={messageId}
                defaultMessage={defaultMessage}
            >
                {(title) => (
                    <i {...props}/>
                )}
            </FormattedMessage>
        );
    }
}
```

Reply:
Thanks for the answer @hmhealey . I indeed forgot about `mount`. My solution wouldn't work with that. 

Also thanks for the better implementation of the `TranslatedIcon`. Do you think it's good to create the `TranslatedIcon` and use it in proper places in this PR? Or maybe in the future?

Reply:
I didn't notice that there's on going conversation here before I merged this PR.  Let's have this improvement on a separate PR.